packages/shorebird_cli/lib/src/commands/release_new/android_release_pipeline.dart

```diff
@@ -0,0 +1,198 @@
+import 'package:mason_logger/mason_logger.dart';
+import 'package:shorebird_cli/src/artifact_builder.dart';
+import 'package:shorebird_cli/src/artifact_manager.dart';
+import 'package:shorebird_cli/src/code_push_client_wrapper.dart';
+import 'package:shorebird_cli/src/commands/release_new/release_new_command.dart';
+import 'package:shorebird_cli/src/commands/release_new/release_pipeline.dart';
+import 'package:shorebird_cli/src/doctor.dart';
+import 'package:shorebird_cli/src/logger.dart';
+import 'package:shorebird_cli/src/platform.dart';
+import 'package:shorebird_cli/src/platform/platform.dart';
+import 'package:shorebird_cli/src/shorebird_android_artifacts.dart';
+import 'package:shorebird_cli/src/shorebird_env.dart';
+import 'package:shorebird_cli/src/shorebird_flutter.dart';
+import 'package:shorebird_cli/src/shorebird_validator.dart';
+import 'package:shorebird_cli/src/third_party/flutter_tools/lib/flutter_tools.dart';
+import 'package:shorebird_cli/src/version.dart';
+import 'package:shorebird_code_push_client/shorebird_code_push_client.dart';
+
+/// {@template android_release_pipeline}
+/// Functions to create an Android release.
+/// {@endtemplate}
+class AndroidReleasePipline extends ReleasePipeline {
+  /// {@macro android_release_pipeline}
+  AndroidReleasePipline({required super.argParser, required super.argResults});
+
+  @override
+  ReleaseType get releaseType => ReleaseType.android;
+
+  Set<Arch> get architectures => (argResults['target-platform'] as List<String>)
+      .map(
+        (platform) => AndroidArch.availableAndroidArchs
+            .firstWhere((arch) => arch.targetPlatformCliArg == platform),
+      )
+      .toSet();
+
+  late bool generateApk = argResults['android-artifact'] as String == 'apk';
+  late bool splitApk = argResults['split-per-abi'] == true;
+
+  @override
+  Future<FileSystemEntity> buildReleaseArtifacts() async {
+    final architectures = (argResults['target-platform'] as List<String>)
+        .map(
+          (platform) => AndroidArch.availableAndroidArchs
+              .firstWhere((arch) => arch.targetPlatformCliArg == platform),
+        )
+        .toSet();
+
+    final flutterVersionString = await shorebirdFlutter.getVersionAndRevision();
+
+    final buildAppBundleProgress = logger
+        .progress('Building app bundle with Flutter $flutterVersionString');
+    await artifactBuilder.buildAppBundle(
+      flavor: flavor,
+      target: target,
+      targetPlatforms: architectures,
+    );
+    buildAppBundleProgress.complete();
+
+    if (generateApk) {
+      final buildApkProgress =
+          logger.progress('Building APK with Flutter $flutterVersionString');
+      await artifactBuilder.buildApk(
+        flavor: flavor,
+        target: target,
+        targetPlatforms: architectures,
+      );
+      buildApkProgress.complete();
+    }
+
+    final projectRoot = shorebirdEnv.getShorebirdProjectRoot()!;
+    try {
+      return shorebirdAndroidArtifacts.findAab(
```

Comment:
IIRC `buildAppbundle` already do this verification, doesn't it?

Reply:
Good catch, fixed!